test/resource_unittest.cc

```diff
@@ -0,0 +1,54 @@
+#include "../src/resource.h"
+#include "../src/json.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(ResourceTest, Type) {
+  std::map<std::string, std::string> m;
+  google::MonitoredResource mr("some_resource", m);
+  EXPECT_EQ("some_resource", mr.type());
+}
+
+TEST(ResourceTest, Labels) {
+  google::MonitoredResource mr("", {{"foo", "bar"}, {"bar", "baz"}});
+  EXPECT_EQ(2, mr.labels().size());
+  EXPECT_EQ("bar", mr.labels().at("foo"));
+  EXPECT_EQ("baz", mr.labels().at("bar"));
+}
+
+TEST(ResourceTest, BasicEquality) {
+  google::MonitoredResource mr1("", {});
+  google::MonitoredResource mr2("", {});
+
+  EXPECT_EQ(mr1, mr2);
+}
+
+TEST(ResourceTest, BasicTypeComparison) {
+  google::MonitoredResource mr1("2", {});
+  google::MonitoredResource mr2("1", {});
+
+  EXPECT_LT(mr1, mr2);
```

Comment:
This one confuses me, mr1's key is 2, how is 2 less than 1?

Reply:
Simple -- it's a bug; the implementation is inverted. :-)

Reply:
Not a bug: "2" < "1" and "b" < "a".  This is confusing, so I added simple examples above the expects to make it more clear.

Reply:
Let's be careful here. The specific implementation of `operator<` for resources *does* have a bug, because `x < y` becomes `x.operator<(y)`, which, in the current implementation would evaluate `y.type_ < x.type_ || (y.type_ == x.type_ && y.labels_ < x.labels_)`, which is clearly wrong.
The test @ACEmilG ran to get `"2" < "1"` is flawed. If you compare `std::string`s, it would behave as expected: `std::string("1") < std::string("2")`. When you compare `char*` values, you're comparing pointers, so the order is essentially random, so that doesn't count.

Reply:
I see, thanks for clarification!

Reply:
Sure. It's funny that ultimately, the behavior of the operator didn't matter as long as it's stable, because all it was needed for was using `MonitoredResource` as a key in `std::map`. But it's good to get it right.